### ISSUE TITLE
[Bug]: Signature fails with Ledger

### ISSUE THREAD

**Issue by atillabirer** [MetaMask/metamask-extension#18232]
### Describe the bug

When we attempt to sign a message using ethersjs and Ledger, signing fails with Network Error in the console and the signature pop-up keeps coming back. The called contract reverts with a hex that basically says InvalidSignature() . Signing normally on browser (without ledger) works normally. Ledger with Trust wallet also works normally.

![image](https://user-images.githubusercontent.com/48114560/226132719-7ab722fb-973d-4254-929a-5d0272fe4ade.png)

### Steps to reproduce

1. Connect ledger to metamask
2. call any transaction that requires signing

### Version

10.25.0

### Build type

None

### Browser

Chrome, Firefox, Brave

### Operating system

MacOS, Linux

### Hardware wallet

Ledger

**Comment by atillabirer** [comment]
> Can you share a code sample that we could run to reproduce the failure you are seeing?

yes sir

```
export default function usePermit2() {
  const permit2 = usePermit2Contract()
  const spender = AGGREGATOR_ADDRESS

  const calcNonces = async (signer) => {
    const array = new Uint32Array(21);
    crypto.getRandomValues(array);
    let nonce = await permit2.getNonce(await signer.getAddress(), array[20])
    if (nonce.eq(0)) {
      for (let i = 0; i < 20; i++) {
        nonce = await permit2.getNonce(await signer.getAddress(), await array[i])
        if (!nonce.eq(0)) {
          break
        }
      }
    }
    return nonce
  }
  const getPermit2Signature = useCallback(
    async (token: string, amount: BigNumberish, signer, library) => {
      try {
        const permit: PermitTransferFrom = {
          permitted: {
            token,
            amount,
          },
          spender: spender,
          // nonce: await permit2.nonceBitmap(signer.getAddress(), 0),
          nonce: await calcNonces(signer),
          deadline: MaxUint256,
        }

        const witness: Witness = {
          witnessTypeName: 'Witness',
          witnessType: { Witness: [{ name: 'user', type: 'address' }] },
          witness: { user: spender },
        }

        const { domain, types, values } = SignatureTransfer.getPermitData(
          permit,
          permit2.address,
          await signer.getChainId(),
          witness
        )
        const signature = await signer._signTypedData(domain, types, values)

        return { signature, permit, witness }
       
      } catch (e) {
        console.log('getPermit2Signature', e)
      }
    },
    [permit2]
  )

  return {
    getPermit2Signature,
  }
}
```
tl;dr using crypto.getandomvalues() to get random number as nonce and then signing permitwitnesstransferfrom from permit2 with that nonce

i would like to mention that doing this works with web3 providers, only fails with ledger.

**Comment by atillabirer** [comment]
handled it this way:

```
        const signature = await signer._signTypedData(domain, types, values)
        //replace 
        let {r,s,v} = utils.splitSignature(signature);
        console.log("rsv:",{r,s,v})
        if(v == 0) v = 27
        if(v == 1) v = 28
        console.log(utils.joinSignature({r,s,v}))
        const joined = utils.joinSignature({r,s,v});


        return { signature: joined, permit, witness }
```
but it would be good if metamask flipped v = 0,1 to v=27,28 when ledger returns the v as 0,1.

**Comment by plasmacorral** [comment]
This was resolved with https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/152
